### PR TITLE
operator DrawerView

### PR DESCRIPTION
**Pull request by imanjra**
## What changes are proposed in this pull request?

Add support for rendering operator prompt in a side drawer.

## How is this patch tested? If it is not, please explain why.

Using a test operator with `resolve_input` code below:

```py
def resolve_input(self, ctx):
    inputs = types.Object()
    inputs.str("name", label="Name", required=True)
    return types.Property(
        inputs, view=types.DrawerView(placement="left", label="Drawer")
    )
```

### Preview

https://github.com/voxel51/fiftyone/assets/25350704/23bbaf1f-22ae-4772-b434-669431fc00ae

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [x] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

Add `DrawerView` prompt style

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
  - Introduced a `Resizable` component allowing horizontal resizing of elements.
  - Added operator prompt areas in various components to enhance user interactions.
  - Implemented new operator prompt components such as `OperatorDrawerPrompt`, `OperatorModalPrompt`, and more for dynamic UI interactions.
  - New `DrawerView` class to specify operator prompts as side drawers.

- **Enhancements**
  - Updated `Resizable` import and behavior in `Sidebar` to improve UI flexibility.
  - Enhanced operator prompt functionalities with new execution, form, and output components.
  - Refined export structure for operator prompt related functionalities in operators package.

- **Bug Fixes**
  - Adjusted operator prompt rendering logic to fix UI inconsistencies.

### CODE REVIEW

**Comment by benjaminpkane** [comment]
Should the expanded sample view (modal) also support the drawer view? Some CSS/HTML needs to be adjusted I think

**Comment by imanjra** [comment]
> Should the expanded sample view (modal) also support the drawer view? Some CSS/HTML needs to be adjusted I think
Yah, good idea. Will extend the support to sample viewer

File: app/packages/operators/src/state.ts [hunk 1]
```diff
@@ -339,7 +345,7 @@ export const useOperatorPrompt = () => {
       operator.isRemote ? RESOLVE_TYPE_TTL : 0,
       { leading: true }
     ),
-    [cachedResolvedInput, setResolvedCtx, ctx]
+    [cachedResolvedInput, setResolvedCtx, ctx, operatorURI]
```
Comment (benjaminpkane) [inline app/packages/operators/src/state.ts:348]: Is memoizing operatorURI required? 🤔 
```suggestion
    [cachedResolvedInput, setResolvedCtx, ctx, operator.uri]
```

**Review by benjaminpkane — APPROVED** [review]
Works great! LGTM 🚀

File: app/packages/operators/src/state.ts (hunk 1 above)
Comment (imanjra) [reply app/packages/operators/src/state.ts:348]: No, it's probably not necessary. Will update it as part of rebase 👍

**Pull request merged**